src/test/java/no/hvl/dat250/h2020/group5/controllers/GuestControllerTest.java

```diff
@@ -0,0 +1,96 @@
+package no.hvl.dat250.h2020.group5.controllers;
+
+import no.hvl.dat250.h2020.group5.entities.Guest;
+import no.hvl.dat250.h2020.group5.repositories.GuestRepository;
+import no.hvl.dat250.h2020.group5.responses.GuestResponse;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class GuestControllerTest {
+
+    @Autowired TestRestTemplate template;
+    @Autowired GuestController guestcontroller;
+    @Autowired GuestRepository guestRepository;
+
+    @LocalServerPort private int port;
+    private URL base;
+    private Guest guest;
+    private List<Guest> guests = new ArrayList<>();
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        guestRepository.deleteAll();
```

Comment:
Tror dette kan bli fikset med @Transactional, istedenfor å slette etter hver test. Beskrevet her https://stackoverflow.com/questions/51036215/spring-h2-test-db-does-not-reset-before-each-test og https://stackoverflow.com/questions/55838140/how-can-i-combine-datajpatest-springboottest-in-one-mvc-application-for-testin  (istedenfor å bruke @DataJPATest som fikser det ellers) :)

Reply:
Testet litt med de ulike annonteringene, og ingen ser ut til å fungere 🤷‍♂️. Fikk du det til?

Reply:
Ja, det virker når jeg bruker @AutoConfigureTestDatabase :)

Reply:
Hmm never mind. Den feilet når jeg la til en test så. 😐